addons/api/addon/serializers/auth-method.js

```diff
@@ -32,7 +32,7 @@ export default class AuthMethodSerializer extends ApplicationSerializer {
     // If an attribute has a `for` option, it must match the
     // record's `type`, else the attribute is excluded
     // from serialization.
-    if (options?.for && !options.for.includes(type)) {
+    if (type !== 'password' && options?.for && !options.for.includes(type)) {
```

Comment:
why do we need this hardcoded check? Can't we make use of `for`?

Reply:
@DhariniJeeva I think the serialize workflow is pushing the password type into the serializeAttribute with no `nestedAttributes`, so when it gets sent through `setPassword` serializer it was getting a null. Fixed